beta-src/src/models/BoardClass.ts

```diff
@@ -0,0 +1,210 @@
+import ConvoyGroupClass from "./ConvoyGroupClass";
+import { IUnit, ITerritory, IContext, ITerrStatus } from "./Interfaces";
+import TerritoryClass from "./TerritoryClass";
+import UnitClass from "./UnitClass";
+
+export default class BoardClass {
+  convoyGroups: ConvoyGroupClass[] = [];
+
+  territories: TerritoryClass[] = [];
+
+  units: UnitClass[] = [];
+
+  constructor(
+    public context: IContext,
+    territories: ITerritory[],
+    terrStatus: ITerrStatus[],
+    units: IUnit[],
+  ) {
+    /**
+     * to reduce time, any territories that is either parent or child are stored and proccessed later in line 61
+     */
```

Comment:
Child territory needs to have parent. Within initial iteration, parent territory may not be instantiated yet. Therefore I'm saving them in temp array and connect them later. Otherwise, we have to re-iterate entire territories and find parent for each child. I think we may use a hash table to make those lookup constant

Reply:
in general comments should add context that the code cannot provide. i would try to explain it better and do not refer to a line number because that will have to be updated or become out of date each time this code is changed.